crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -12,94 +12,139 @@ pub use bevy_ecs_macros::WorldQuery;
-/// allowing you to query for components immutably and mutably accordingly.
-///
-/// See [`Query`](crate::system::Query) for a primer on queries.
-///
-/// # Basic [`WorldQuery`]'s
-///
-/// Here is a small list of the most important world queries to know about where `C` stands for a
-/// [`Component`] and `WQ` stands for a [`WorldQuery`]:
-/// - `&C`: Queries immutably for the component `C`
-/// - `&mut C`: Queries mutably for the component `C`
-/// - `Option<WQ>`: Queries the inner [`WorldQuery`] `WQ` but instead of discarding the entity if the world
-///     query fails it returns [`None`]. See [`Query`](crate::system::Query).
-/// - `(WQ1, WQ2, ...)`: Queries all contained world queries allowing to query for more than one thing.
-///     This is the `And` operator for filters. See [`Or`].
-/// - `ChangeTrackers<C>`: See the docs of [`ChangeTrackers`].
-/// - [`Entity`]: Using the entity type as a world query will grant access to the entity that is
-///     being queried for. See [`Entity`].
-///
-/// Bevy also offers a few filters like [`Added`](crate::query::Added), [`Changed`](crate::query::Changed),
-/// [`With`](crate::query::With), [`Without`](crate::query::Without) and [`Or`].
-/// For more information on these consult the item's corresponding documentation.
-///
-/// [`Or`]: crate::query::Or
-///
-/// # Derive
-///
-/// This trait can be derived with the [`derive@super::WorldQuery`] macro.
-///
-/// You may want to implement a custom query with the derive macro for the following reasons:
-/// - Named structs can be clearer and easier to use than complex query tuples. Access via struct
-///   fields is more convenient than destructuring tuples or accessing them via `q.0, q.1, ...`
-///   pattern and saves a lot of maintenance burden when adding or removing components.
-/// - Nested queries enable the composition pattern and makes query types easier to re-use.
-/// - You can bypass the limit of 15 components that exists for query tuples.
+/// Types that can be fetched from a [`World`] using a [`Query`].
+///
+/// There are many types that natively implement this trait:
+///
+/// - **Component references.**
+///   Fetches a component by reference (immutably or mutably).
+/// - **`WorldQuery` tuples.**
+///   If every element of a tuple implements `WorldQuery`, then the tuple itself also implements the same trait.
+///   This enables a single `Query` to access multiple components and filter over multiple conditions.
+///   Due to the current lack of variadic generics in Rust, the trait has been implemented for tuples from 0 to 15 elements,
+///   but nesting of tuples allows infinite `WorldQuery`s.
+/// - **Component filters.**
+///   [`With`] and [`Without`] filters can be applied to check if the queried entity contains or not a particular component.
+/// - **Change detection filters.**
+///   [`Added`] and [`Changed`] filters can be applied to detect component changes to an entity.
```

Comment:
Is the `Removed` filter something that should be mentioned here, or does that work differently?

Reply:
Do we have a `Removed` filter? I [cannot find] it... I also remember having something like that in the past, but I think it got removed (or is it Mandela effect 😛 )

[cannot find]: https://dev-docs.bevyengine.org/bevy/index.html?search=Removed

Reply:
Seems we can detect it, but that it is specifically _not_ a filter:
> Removal detection is special. This is because, unlike with [change detection](https://bevy-cheatbook.github.io/programming/change-detection.html), the data does not exist in the ECS anymore (obviously), so Bevy cannot keep tracking metadata for it.
> Nevertheless, being able to respond to removals is important for some applications, so Bevy offers a limited form of it.

https://bevy-cheatbook.github.io/programming/removal-detection.html

So to answer my own question: Yes, it works differently.